src/diffpy/utils/diffraction_objects.py

```diff
@@ -394,7 +395,7 @@ def on_xtype(self, xtype):
         elif xtype.lower() in DQUANTITIES:
             return self.on_d()
         else:
-            warnings.warn(_xtype_wmsg(xtype))
+            raise ValueError(_xtype_wmsg(xtype))
```

Comment:
Is it better if we raise an error here?

Reply:
I am not sure I understand....we are raising an error...